app/controllers/api/groups_controller.rb

```diff
@@ -142,11 +142,34 @@ def update_marks
 
     def set_mark_by_criteria(criteria, mark_to_change)
```

Comment:
It's okay, it's really an instance of the flexible/rubric criterion inheritance not being implemented in a nice way.